src/platform/platform.cpp

```diff
@@ -680,64 +680,30 @@ void DebugPrint(const char *fmt, ...) {
 #endif
 
 //-----------------------------------------------------------------------------
-// Temporary arena, on Windows.
+// Temporary arena.
 //-----------------------------------------------------------------------------
 
-#if defined(WIN32)
-
-static HANDLE TempArena = NULL;
-
-void *AllocTemporary(size_t size)
-{
-    if(!TempArena)
-        TempArena = HeapCreate(0, 0, 0);
-    void *ptr = HeapAlloc(TempArena, HEAP_ZERO_MEMORY, size);
-    ssassert(ptr != NULL, "out of memory");
-    return ptr;
-}
+struct MimallocHeap {
+    mi_heap_t *heap = mi_heap_new();
 
-void FreeAllTemporary()
-{
-    HeapDestroy(TempArena);
-    TempArena = NULL;
-}
-
-#endif
-
-//-----------------------------------------------------------------------------
-// Temporary arena, on Linux.
-//-----------------------------------------------------------------------------
-
-#if !defined(WIN32)
-
-struct ArenaChunk {
-    ArenaChunk *next;
+    ~MimallocHeap() {
+        if(heap)
```

Comment:
This is always true now

Reply:
It might not be, considering that `mi_heap_new()` heads with 
```c
mi_heap_t* mi_heap_new(void) {
  mi_heap_t* bheap = mi_heap_get_backing();
  mi_heap_t* heap = mi_heap_malloc_tp(bheap, mi_heap_t);  // todo: OS allocate in secure mode?
  if (heap==NULL) return NULL;
```

Seeing as `mi_heap_malloc()` (and `mi_heap_malloc_small()` similarly) starts with
```c
extern inline mi_decl_restrict void* mi_heap_malloc(mi_heap_t* heap, size_t size) mi_attr_noexcept {
  if (mi_likely(size <= MI_SMALL_SIZE_MAX)) {
    return mi_heap_malloc_small(heap, size);
  }
  else {
    mi_assert(heap!=NULL);
```
and potentially aborts under us, I opted for a `ssassert()` in the constructor instead.

Reply:
Ah yeah good catch!